low-level/sensors/impl/centos_7/systemd_watchdog.py

```diff
@@ -1415,9 +1415,36 @@ def _get_drive_fault_info(self, path):
             cmd = f"sudo smartctl -d scsi -A {self._drive_by_device_name[path]}"
         else:
             cmd = f"sudo smartctl -A {self._drive_by_device_name[path]}"
-        response, _ = self._run_command(cmd)
+        response, _, _ = self._run_command(cmd)
         return response
 
+    def _is_local_drive(self, object_path):
+        """
+        Detect Node server local drives using Hdparm tool.
+        Hdparm tool give information only for node drive.
+        For external JBOD/virtual drives it will give output as:
+        "SG_IO: bad/missing sense data, sb[]:  70 00 05 00 00 00
+        00 0e 00 00 00 00 20 00 00 00 00 00 00 00 00 00 00 00 00 00 00
+        00 00 00 00 00\n HDIO_GET_IDENTITY failed: Invalid argument"
+        Hdparm does not have support for NVME drives, for this drives it gives o/p as:
+        "failed: Inappropriate ioctl for device"
+        """
+        ENCL_DISK_ERR = "SG_IO: bad/missing sense data, sb[]:  70 00 05" + \
+                        " 00 00 00 00 0e 00 00 00 00 20 00 00 00 00 00 00 00 00" + \
+                        " 00 00 00 00 00 00 00 00 00 00 00\n HDIO_GET_IDENTITY failed: Invalid argument"
+
+        drive_name = self._drive_by_device_name[object_path]
+        cmd = f'sudo hdparm -i {drive_name}'
+        _, err, retcode = self._run_command(cmd)
+        if retcode == 0:
+            return True
+        else:
+            logger.debug(f"SystemdWatchdog, _is_local_drive: Error for drive {drive_name}, ERROR: {err}")
+            if ENCL_DISK_ERR not in err:
+                return True

```

Comment:
in case of different for local drive, this check would fail still, so not full proof. add a TODO mentioning same

Reply:
Done.